src/ui/ui_settings/ui_settings_service_for_request.js

```diff
@@ -0,0 +1,20 @@
+import { uiSettingsServiceFactory } from './ui_settings_service_factory';
+
+const BY_REQUEST_CACHE = new WeakMap();
+
+export function getUiSettingsServiceForRequest(server, request, readInterceptor) {
+  if (BY_REQUEST_CACHE.has(request)) {
```

Comment:
  Sure, works for me.

> o know what a "valid" cache looks like, and just determines the existence of the cache at all, meaning that the similar caching logic will look exactly the same even if the cache contents can be "falsy" or even null/undefined.

[Spec says](http://www.ecma-international.org/ecma-262/6.0/#sec-map.prototype.get) it's `undefined` if value is not presented in the map. So yeah it should be `if (uiSettingsServices !== undefined) {` :) But at the same it allows you to add `undefined` as a value, so I tend to agree with you on that.

> I also can't imagine that there is any perceptible performance difference

Even though `has-get` pair basically means two `get` calls I agree in our case it shouldn't  be a big deal.